docs/applications.md

```diff
@@ -17,8 +17,18 @@ limitations under the License.
 
 # Fluo Applications
 
-Once you have Fluo installed and running on your cluster, you can now run Fluo applications which
-consist of clients and observers.
+Once you have Fluo installed and running on your cluster, you can run Fluo applications consisting
+of clients and observers.  A client is a user started process that uses the Fluo API.  An Observer
+is user provided code run by Fluo Workers in response to notifications.  Notifications are set by
```

Comment:
Could link to architecture.md here if users want to better understand components of an application.  This text could be incorporated into architecture.md to improve it.